tests/test_max_flow.py

```diff
@@ -0,0 +1,160 @@
+import unittest
+
+import numpy as np
+
+try:
+    import networkx as nx
+except ImportError:
+    nx = None
+
+import gurobi_optimods.datasets as datasets
+from gurobi_optimods.max_flow import max_flow
+
+from .test_graph_utils import (
+    check_solution_networkx,
+    check_solution_pandas,
+    check_solution_scipy,
+)
+from .test_min_cost_flow import (
+    load_graph2_networkx,
+    load_graph2_pandas,
+    load_graph2_scipy,
+)
+
+
+class TestMaxFlow(unittest.TestCase):
+    def setUp(self):
+        self.expected_max_flow = 3.0
+
+    def test_infeasible(self):
```

Comment:
Should this be infeasible? There is no demand, so "unsatisfiable flows" doesn't make sense. Really I think this should return max flow of zero.

Equivalent test cases are also needed for the other input formats, since the formulation codes are independent.